File: DoitMission_08/Doit08_Java/app/src/main/java/com/example/doit08_java/ProductActivity.java
package com.example.doit08_java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class ProductActivity extends AppCompatActivity {

    private Intent intent;
    private Button pMenu;
    private Button pLogin;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_product);
        intent = getIntent();

        pMenu = findViewById(R.id.productmenu);
        pLogin = findViewById(R.id.productlogin);

        pMenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                setResult(RESULT_OK,intent);
                finish();
            }
        });

        pLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String cutData = intent.getStringExtra(MenuActivity.MENU_REQUEST);
                intent = new Intent(getApplicationContext(),MainActivity.class);
                intent.putExtra(MenuActivity.MENU_REQUEST,cutData);
                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                startActivityForResult(intent,302);
            }
        });
    }


    @Override
    public void onBackPressed() {

    }
}

File: DoitMission_08/Doit08_Kotlin/app/src/main/java/com/example/doit08_kotlin/ProductActivity.kt
package com.example.doit08_kotlin

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_product.*
import org.jetbrains.anko.startActivity

class ProductActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_product)

        val intent = intent

        productmenu.setOnClickListener {
            setResult(Activity.RESULT_OK,intent)
            finish()
        }

        productlogin.setOnClickListener {
            val inten1 = Intent(this, MainActivity::class.java)
            inten1.putExtra("data",intent.getStringExtra("data"))
            inten1.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
            startActivity(inten1)
        }
    }

    override fun onBackPressed() {

    }
}
